head/src/main/kotlin/io/qalipsis/core/head/report/chart/ChartServiceImpl.kt

```diff
@@ -0,0 +1,329 @@
+/*
+ * QALIPSIS
+ * Copyright (C) 2022 AERIS IT Solutions GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package io.qalipsis.core.head.report.chart
+
+import io.qalipsis.api.lang.alsoWhenNull
+import io.qalipsis.api.report.TimeSeriesAggregationResult
+import io.qalipsis.core.head.jdbc.entity.DataSeriesEntity
+import jakarta.inject.Singleton
+import org.jfree.chart.ChartFactory
+import org.jfree.chart.JFreeChart
+import org.jfree.chart.LegendItem
+import org.jfree.chart.LegendItemCollection
+import org.jfree.chart.LegendItemSource
+import org.jfree.chart.StandardChartTheme
+import org.jfree.chart.axis.DateAxis
+import org.jfree.chart.axis.NumberAxis
+import org.jfree.chart.block.BlockBorder
+import org.jfree.chart.plot.XYPlot
+import org.jfree.chart.renderer.xy.XYSplineRenderer
+import org.jfree.chart.title.LegendTitle
+import org.jfree.data.xy.XYSeries
+import org.jfree.svg.SVGGraphics2D
+import org.jfree.svg.SVGUtils
+import org.jfree.ui.RectangleEdge
+import java.awt.Color
+import java.awt.Font
+import java.awt.Rectangle
+import java.io.BufferedInputStream
+import java.io.FileInputStream
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.Random
+
+
+/**
+ * Custom implementation of [ChartService] to plot time-series charts.
+ *
+ * @author Francisca Eze
+ */
+@Singleton
+internal class ChartServiceImpl(
+    private val lineStyleGenerator: LineStyleGenerator
+) : ChartService {
+    private lateinit var dataSeriesDrawingConfigurations: MutableMap<String, DataSeriesDrawingConfiguration>
+    private lateinit var campaignKeyToLineMap: MutableMap<String, Int>
+
+    override fun buildChart(
+        data: Map<String, List<TimeSeriesAggregationResult>>, dataSeries: Collection<DataSeriesEntity>, index: Int
+    ): Path {
+        dataSeriesDrawingConfigurations = mutableMapOf()
+        //Re-arrange the data to build time series pairs and other data structures needed to render and style the charts.
```

Comment:
```suggestion
        // Re-arrange the data to build time series pairs and other data structures needed to render and style the charts.
```
Please add this space everywhere. It helps a lot to read.